roofit/roofitcore/test/testRooFuncWrapper.cxx

```diff
@@ -84,6 +86,64 @@ TEST(RooFuncWrapper, GaussianNormalizedHardcoded)
    EXPECT_NEAR(getNumDerivative(gauss, sigma, normSet), dMyGauss[2], 1e-8);
 }
 
+TEST(RooFuncWrapper, NllWithObservables)
+{
+   using namespace RooFit;
+
+   auto inf = std::numeric_limits<double>::infinity();
+   RooRealVar x("x", "x", 0, -inf, inf);
+   RooRealVar mu("mu", "mu", 0, -10, 10);
+   RooRealVar sigma("sigma", "sigma", 2.0, 0.01, 10);
+   RooGaussian gauss{"gauss", "gauss", x, mu, sigma};
+
+   RooArgSet normSet{x};
+
+   std::size_t nEvents = 10;
+   std::unique_ptr<RooDataSet> data{gauss.generate(x, nEvents)};
+   std::unique_ptr<RooAbsReal> nllRef{gauss.createNLL(*data)};
+
+   RooArgSet parameters;
+   gauss.getParameters(data->get(), parameters);
+
+   RooArgSet observables;
+   gauss.getObservables(data->get(), observables);
```

Comment:
I think it's a bit inconvienient that the caller has to figure out what are parameters and what are observables.

Can you do this in the `RooFuncWrapper` constructor instead?

Then it would just be called as `RooFuncWrapper<> nllFunc("myNLL", "myNLL", func.str(), parameters, observables, data.get());`, with the data as an optional pointer. And when it's a `nullptr`, all variables are parameters.

What do you think?

Reply:
Hmm, sure, but don't they need the pdf (of what the rooFunc represents) to get the params from? That seems sort of counterintuitive to me because you would not expect the usage to be like that.

On the 'pass as a pointer' argument, I did not do that because I did not see this being done anywhere else and just wanted to be consistent.

Reply:
Sorry, I see my comment was really not clear. I meant you can do the splitting between parameters and observables in the constructor, and pass all variables like: `RooFuncWrapper<> nllFunc("myNLL", "myNLL", func.str(), variables, data);`

Anyway, let's leave it as you suggest now. It's fine too and probably safer for now, because the order of variables and parameters is clearly defined.

Reply:
For the pass by pointer: okay you can also do something else if you like, but always taking a `RooAbsData` in the constructor is definitely not good because it forces you to pass a dummy `RooDataSet` object. Just try to avoid that please.

Reply:
I can change it to be a pointer. That's not a problem.